### PR TITLE
Show position on config panel

### PR DESCRIPTION
**Pull request by robert00s**
Actually we don't know on which panel we are (in config panel). After this change we have inverted button to show which panel we use.
Before:
screenshot 2017-09-09 18-45-54
After:
screenshot 2017-09-09 18-47-16

### CODE REVIEW

**Comment by poire-z** [comment]
Would be nice if both top and bottom menus had a similar way of showing the active icon.
For the top menu, the horizontal line is cut under the active icon.
(I would personally prefer the way the top menu does it, as, on my Kobo, a partial refresh, when a big block of black disappears, leaves the new text at its place a bit lighter than the other text that was not under black, so I guess i would feel the need to do a full refresh after closing bottom menu to restore balance :)
(Haven't looked at how possible or hard it is.)

**Comment by Frenzie** [comment]
> For the top menu, the horizontal line is cut under the active icon.

Not just that. It also has the lines on the side. ;-)

I'll have to try this on my device, but I suspect that rather than a plain invert I would prefer a dark gray like in the MultiSelects (or whatever they're called) in the menu itself. If we choose to go for this style, that is.

**Comment by robert00s** [comment]
Better?
screenshot 2017-09-10 00-16-28

**Comment by KenMaltby** [comment]
I think the matching grey indicator is a little more consistent and obvious, but the last approach also is consistent with the top menu treatment.

File: frontend/ui/widget/configdialog.lua [hunk 1]
```diff
@@ -426,20 +432,68 @@ function MenuBar:init()
...
+    }
+    local spacing_width = math.ceil((Screen:getWidth() - icons_width) / (#menu_items+1))
     local spacing = HorizontalSpan:new{
-        width = (Screen:getWidth() - icons_width) / (#menu_items+1)
+        width = spacing_width,
+    }
+    local spacing_line = LineWidget:new{
+        dimen = Geom:new{
+            w = spacing_width,
+            h = 2,
```
Comment (Frenzie) [inline frontend/ui/widget/configdialog.lua]: Shouldn't this one be `icon_sep_width`?

File: frontend/ui/widget/configdialog.lua [hunk 2]
```diff
@@ -426,20 +432,68 @@ function MenuBar:init()
...
+        }
+    }
+    local sep_line = LineWidget:new{
+        dimen = Geom:new{
+            w = icon_sep_width,
+            h = 2,
```
Comment (Frenzie) [inline frontend/ui/widget/configdialog.lua]: And this one.

File: frontend/ui/widget/configdialog.lua [hunk 3]
```diff
@@ -417,6 +417,7 @@ function MenuBar:init()
     local menu_items = {}
     local icons_width = 0
     local icons_height = 0
+    local line_thickness = 2
```
Comment (Frenzie) [inline frontend/ui/widget/configdialog.lua]: Like I said though, shouldn't it also be given the scaleBySize treatment? :-)

**Comment by Frenzie** [comment]
Looks okay but we're still missing scaling on the lines on the very top and very bottom. Could you do those as well? :-)

screenshot_2017-09-10_21-30-08-fs8

**Comment by Frenzie** [comment]
@poire-z Also I don't think those ImageWidgets are scaling properly.

**Comment by poire-z** [comment]
I don't have a monitor big enough to reproduce what you show :)
IconButton have a `scale_for_dpi = true` parameter. Do you get different results if you add it and set to false or true in
https://github.com/koreader/koreader/blob/a720eaceb50f32465e22d8210638cc6d2541f7d7/frontend/ui/widget/configdialog.lua#L415-L417

Also, I don't know much about dpi mathematics, but when I checked the following, I think I remember it was multiplying by 2 at most:
https://github.com/koreader/koreader/blob/a720eaceb50f32465e22d8210638cc6d2541f7d7/frontend/ui/widget/imagewidget.lua#L178-L182

**Comment by Frenzie** [comment]
@poire-z I get what looks like good results by removing the flooring.

```
--- a/frontend/ui/widget/imagewidget.lua
+++ b/frontend/ui/widget/imagewidget.lua
@@ -177,9 +177,9 @@ function ImageWidget:_render()
 
     -- scale_for_dpi setting: update scale_factor (even if not set) with it
     if self.scale_for_dpi then
+        local size_scale = math.min(Screen:getWidth(), Screen:getHeight())/600
         local dpi_scale = Screen:getDPI() / 167
-        -- rounding off to power of 2 to avoid alias with pow(2, floor(log(x)/log(2))
-        dpi_scale = math.pow(2, math.max(0, math.floor(math.log(dpi_scale)/0.69)))
+        dpi_scale = math.pow(2, math.max(0, math.log((size_scale+dpi_scale)/2)/0.69))
         if self.scale_factor == nil then
             self.scale_factor = 1
         end
diff --git a/test b/test
```

### Before

screenshot_2017-09-10_22-45-17-fs8

### After

screenshot_2017-09-10_22-45-00-fs8

**Comment by poire-z** [comment]
Well, I don't really get dpi arithmetics :)
But your changes looks the same on the emulator, but are way to big on my kobo:
<kbd>reader_2017-sep-10_230634</kbd>
(it looks more monstruous on the device that it does on this reduced screenshot)

**Comment by robert00s** [comment]
@Frenzie 
Could you check now?
I wonder if this line:
https://github.com/koreader/koreader/blob/a9c1ed24b8b95d6b7eb3b91d85d4079f43a1c910/frontend/ui/widget/container/framecontainer.lua#L30
shouldn't be 
`bordersize = Screen:scaleBySize(2),`
?

**Comment by Frenzie** [comment]
Yes, that is good. But like you said, don't do it in ConfigDialog and fix FrameContainer instead. 👍

File: frontend/ui/widget/container/framecontainer.lua [hunk 4]
```diff
@@ -18,16 +18,18 @@ Example:
 
 --]]
 
-local WidgetContainer = require("ui/widget/container/widgetcontainer")
-local Geom = require("ui/geometry")
 local Blitbuffer = require("ffi/blitbuffer")
+local Device = require("device")
+local Geom = require("ui/geometry")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local Screen = Device.screen
```
Comment (Frenzie) [inline frontend/ui/widget/container/framecontainer.lua]: Nitpick, but if you only need screen then local Screen = require("device").screen is fine. ;-)

File: frontend/ui/widget/container/framecontainer.lua [hunk 5]
```diff
@@ -18,16 +18,18 @@ Example:
...
 
 local FrameContainer = WidgetContainer:new{
     background = nil,
     color = Blitbuffer.COLOR_BLACK,
     margin = 0,
     radius = 0,
-    bordersize = 2,
+    bordersize = Screen:scaleBySize(2),
     padding = 5,
```
Comment (Frenzie) [inline frontend/ui/widget/container/framecontainer.lua]: This one too please? 👍

**Comment by Frenzie** [comment]
Thanks!

**Comment by poire-z** [comment]
Noticed 2 things: icons move a few pixels left or right upon selection/unselection, and the selected icon looks to me a bit tight between its 2 vertical lines.
I'm having a try at spreading the spacing between icons into some padding between icon and vertical lines.

**Comment by Frenzie** [comment]
Could you make sure the padding is touchable? The top menu also wants that treatment. ;-)

**Pull request merged**